tt_metal/llrt/llrt.cpp

```diff
@@ -49,13 +50,15 @@ ll_api::memory get_risc_binary(string const &path, uint32_t riscv_id, PackSpans
     if (inserted) {
       // We're the first with PATH. Create and insert.
       lock.unlock();
-      auto *ptr = new ll_api::memory(path);
+      auto *ptr = new ll_api::memory(path, relo_type);
 
       // TODO: pass pack_spans into reader, generate text/data sizes
       // from segment sizes and pack there
-      if (pack_spans == PackSpans::PACK) {
+      if (span_type == ll_api::memory::PackSpans::PACK) {
           uint64_t data_start = MEM_LOCAL_BASE;
-          uint64_t text_start = processor_to_fw_base_addr[riscv_id];
+          uint64_t text_start = (relo_type == ll_api::memory::Relocate::XIP) ?
+              0 :
+              processor_to_fw_base_addr[riscv_id];
```

Comment:
It's unclear to me why the segment's address coming out of tt_elffile isn't sufficient here?

Reply:
Maybe it is, I'll check.  Some of this is legacy (reverse lookup of the address)